src/azul/indexer/document.py

```diff
@@ -274,13 +292,30 @@ def to_index(self, value: T) -> T:
     def from_index(self, value: T) -> T:
         return value
 
+    @property
+    def json_type(self):
+        if self.n == JSON:
+            return make_type(str)
+        else:
+            return make_type(self.n)
+
 
-pass_thru_str: PassThrough[str] = PassThrough(es_type='string')
-pass_thru_int: PassThrough[int] = PassThrough(es_type='long')
-pass_thru_bool: PassThrough[bool] = PassThrough(es_type='boolean')
+pass_thru_str: PassThrough[str] = PassThrough(str, es_type='string')
+pass_thru_int: PassThrough[int] = PassThrough(int, es_type='long')
+pass_thru_bool: PassThrough[bool] = PassThrough(bool, es_type='boolean')
 # FIXME: change the es_type for JSON to `nested`
 #        https://github.com/DataBiosphere/azul/issues/2621
-pass_thru_json: PassThrough[JSON] = PassThrough(es_type=None)
+pass_thru_json: PassThrough[JSON] = PassThrough(JSON, es_type=None)
+
+
+class Range(PassThrough[JSON]):
+
+    @property
+    def json_type(self):
+        return make_type(int)
```

Comment:
It's not obvious why this is `int`. Either it's a bug or sit should be explained.